lib/Graph/Graph.cpp

```diff
@@ -3037,6 +3037,26 @@ SaveNode *glow::getOutputSave(Function *F, Placeholder *PH) {
   return nullptr;
 }
 
+Node *glow::recursiveClone(Function *newF, Node *node, NodeMap &currToNew) {
```

Comment:
According to my understanding, a constant operation's inputs could be an "constant" or an "splat" node (An op can become a constant op when its all inputs are replaced with constant or splat nodes). So only if the input is an "splat" node, we need to copy the input. So maybe the "recursive" is not necessary? Just check if the input type is "splat", then copy the input as well.

Reply:
@beicy `recursiveClone` does not copy `Storage` nodes. And `Constant` is one of such storage nodes. So, constant won't be copied.

Also, keep in mind that `recursiveClone` is a helper function that is used not only for constant operations evaluations, but in some other places as well. There is needs to be able to copy arbitrary nodes.